### ISSUE TITLE
Plotting exercise: proportions and factors

### ISSUE THREAD

**Issue by lconover1** [USF-Psych-DataSci-2020/Discussion#2]
I am having a hard time figuring out 6a for the classwork from last class.  
I can get the x-variable to not appear continuous:
`mtcars %>% 
  mutate(transmission = if_else(am == 0, "automatic", "manual")) %>% 
  ggplot(aes(x = factor(cyl))) +
  geom_bar(aes(color = "black", fill = transmission), position = "dodge")`
  
I can get the y-variable to be a proportion rather than a count:
`mtcars %>% 
  mutate(transmission = if_else(am == 0, "automatic", "manual")) %>% 
  ggplot(aes(cyl)) +
  geom_bar(aes(y = ..prop.., color = "black", fill = transmission), position = "dodge")`

However, when I try to do both at once, all of my bars just end up the same height:
`mtcars %>% 
  mutate(transmission = if_else(am == 0, "automatic", "manual")) %>% 
  ggplot(aes(x = factor(cyl))) +
  geom_bar(aes(y = ..prop.., color = "black", fill = transmission), position = "dodge")`

The hint said to change the variable, but the attempts I made with the "mutate" function and cyl just lead to the same outcome as combining factor and ..prop.. (unfortunately, I did not save exactly what I did there.  I know, I know, I should have.)

Anyone have any advice?

**Comment by wesley4546** [comment]
Hey,

So when changing the x axis to only include 4, 6, and 8 we need to tell ggplot that the cyl column is a factor not a numerical value.

(We know its a numerical value by checking the structure of the data set ` str(mtcars)`). 

We change it to a factor by the `factor()` function:

`ggplot(aes(x=factor(variable)))`

When it comes to making the y variable a proportion and not a count, in the geom_bar we make position = "fill":

`geom_bar(aes(color = variable), position = "fill")`

Hope I helped!

**Comment by lconover1** [comment]
> Hey,
> 
> So when changing the x axis to only include 4, 6, and 8 we need to tell ggplot that the cyl column is a factor not a numerical value.
> 
> (We know its a numerical value by checking the structure of the data set ` str(mtcars)`).
> 
> We change it to a factor by the `factor()` function:
> 
> `ggplot(aes(x=factor(variable)))`
> 
> When it comes to making the y variable a proportion and not a count, in the geom_bar we make position = "fill":
> 
> `geom_bar(aes(color = variable), position = "fill")`
> 
> Hope I helped!

No, unfortunately not.  The first suggestion regarding factor is what I was already using:  

> I can get the x-variable to not appear continuous:
> mtcars %>% mutate(transmission = if_else(am == 0, "automatic", "manual")) %>% ggplot(aes(x = factor(cyl))) + geom_bar(aes(color = "black", fill = transmission), position = "dodge")

The second suggestion regarding position puts "proportion" on the y-axis label, but it does not allow the bars to be set side-by-side as the rest of the problem requests.

**Comment by wesley4546** [comment]
I see.

Yeah I also am confused on that specific part of the assignment. (Step 4 of 6(a)). I also made a issue regarding that. I'm not sure how we can show the proportion of transmission by using the available code other than to manipulate the data in a way that allows us to take the `sum()` of the count of manual and automatic.
